resolwe_bio/processes/samtools/samtools_bedcov.py

```diff
@@ -0,0 +1,184 @@
+"""Samtools bedcov."""
+
+from io import StringIO
+from pathlib import Path
+
+import pandas as pd
+from plumbum import TEE
+
+from resolwe.process import (
+    BooleanField,
+    Cmd,
+    DataField,
+    FileField,
+    GroupField,
+    IntegerField,
+    ListField,
+    Process,
+    SchedulingClass,
+    StringField,
+)
+
+
+def normalize_bedcov_output(output: str) -> pd.DataFrame:
+    """Calculate mean coverage over regions specified in samtools bedcov output.
+
+    Samtools bedcov returns sum of read coverages across all bases in bed file regions.
+    This function normalizes the values with the length of individual regions.
+
+    Input columns : [chr, start, stop, {coverage_sum1, coverage_sum2, ...}]
+    Output columns : [chr, start, stop, {mean_coverage1, mean_coverage2, ...}]
+    """
+    coverage_df = pd.read_csv(
+        StringIO(output),
+        sep="\t",
+        header=None,
+    )
+
+    start = coverage_df.iloc[:, 1]
+    end = coverage_df.iloc[:, 2]
+    sum_columns = coverage_df.iloc[:, 3:]
+
+    for column_name in sum_columns.columns:
+        coverage_df[column_name] = sum_columns[column_name] / (end - start)
+
+    return coverage_df
+
+
+class SamtoolsBedcov(Process):
+    """Samtools bedcov.
+
+    Reports the total read base count (i.e. the sum of per base read depths)
+    for each genomic region specified in the supplied BED file.
+    The regions are output as they appear in the BED file and are 0-based.
+    Counts for each alignment file supplied are reported in separate columns.
+    The output is tab-delimited with the first three columns being the chromosome,
+    start and end positions of the region, and the remaining columns
+    being the sum read base counts for each alignment.
+
+    For more information about samtools bedcov, click
+    [here](https://www.htslib.org/doc/samtools-bedcov.html).
+
+    """
+
+    slug = "samtools-bedcov"
+    process_type = "data:alignment:bam:samtools"
+    name = "Samtools bedcov"
+    requirements = {
+        "expression-engine": "jinja",
+        "executor": {
+            "docker": {"image": "public.ecr.aws/genialis/resolwebio/rnaseq:6.0.0"},
+        },
+        "resources": {
+            "cores": 2,
+            "memory": 16384,
+        },
+    }
+    category = "Samtools"
+    data_name = "{{ bam|name|default('?') }}"
+    version = "1.0.0"
+    entity = {"type": "sample"}
+    scheduling_class = SchedulingClass.BATCH
+
+    class Input:
+        """Input fields for SamtoolsBedcov."""
+
+        bams = ListField(
```

Comment:
Each BAM from `bams` input is one sample. Defining entity = {"type": "sample"} will result in a new sample being crated, as the inputs are already part of individual/separate samples. 
If the goal is to attach this data to the existing samples, then each BAM should be processed separately (remove ListField). If there is a need for multisample bedcov process, this should be a separate analysis.

Reply:
Ok, we will then use it per sample.